chore: add CHANGELOG.md (#1084)
terramate-io/terramate: CHANGELOG.md

# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## Versioning

This Solution adheres to the principles of [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

Given a version number `MAJOR.MINOR.PATCH`, we increment the:

1. `MAJOR` version when we make incompatible changes,
2. `MINOR` version when we add functionality in a backward compatible manner, and
3. `PATCH` version when we make backward compatible bug fixes.

### Backward compatibility in `0.0.z` and `0.y.z` version

> - Backward compatibility in versions `0.0.z` is **not guaranteed** when `z` is increased.

Reviewer: We left this stage ;)

```suggestion
```

- Backward compatibility in versions `0.y.z` is **not guaranteed** when `y` is increased.

> ## 0.4.1-dev

Reviewer: ```suggestion
## Unreleased
```

### Added

> - Add support for glob/wildcards in the `import.source`.
> - Sending of pull_request information when running locally.
> - Make cloud host and IDP configurable.

Reviewer: ```suggestion
- Add support for globs in the `import.source` attribute to import multiple files at once.
```

### Fixed

> - allow comma in the --tags of the create command.

Reviewer: ```suggestion
- Allow to specify multiple tags separated by comma when using `terramate create --tags` command.
```

> - The `commit_sha` should be always sent when creating a TMC deployment.

Reviewer: ```suggestion
```

## 0.4.0

### Added

> - Introduce the `terramate create --all-terraform` command.

Reviewer: ```suggestion
- Introduce the `terramate create --all-terraform` command that can be used to initialize plain terraform projects easily with a Terramate Stack configuration.
  Every directory containing Terraform Files that configure either a `terraform.backend {}` block or a `provider {}` block will be initialized.
```

> - Promote experimental `ensure-stack-id` to `create --ensure-stack-ids`.

Reviewer: ```suggestion
- Introduce `terramate create --ensure-stack-ids` to add a UUID as stack ID where `stack.id` is not set. (Promoted from `experimental ensure-stack-id`)
```

### Changed

> - BREAKING CHANGE: Make stack.id case insensitive.
> - The `terramate-ls` distributed alongside `terramate`.

Reviewer: ```suggestion
- BREAKING CHANGE: Make `stack.id` case insensitive. If you used case to identify different stacks, the stack ID need to be adjusted. If you use the recommended UUIDs nothing has to be done.
- We now distribute the Terramate Language Server `terramate-ls` alongside `terramate` to be able to pick up changes faster for IDEs.
```

## 0.3.1

### Added

> - Introduce the `terramate experimental cloud info` command.
> - Introduce the `--cloud-sync-deployment` flag.
> - Set repository=local if it points to a local filesystem path.
> - Send Pull Request information when synchronizing the deployment.
> - Send detailed ReviewRequest information.
> - Introduce the `terramate experimental ensure-stack-id` command.

Reviewer: ```suggestion
- Introduce the `terramate experimental cloud info` command.
- Introduce the `--cloud-sync-deployment` flag (experimental).
- Introduce the `terramate experimental ensure-stack-id` command.
```

### Fixed

> - Crash when having `generate_hcl` with no `content` block.

Reviewer: ```suggestion
- Fix a crash when `generate_hcl {}` has no `content {}` block defined.
```

## 0.3.0

### Added

- Introduced the `terramate experimental cloud login` command.

### Fixed

> - File descriptor leak in the configuration tree load

Reviewer: ```suggestion
- Fix a file descriptor leak when loading the configuration tree.
```
